improver/psychrometric_calculations/total_precipitable_water.py

```diff
@@ -0,0 +1,169 @@
+        """
+        delta_p = -np.diff(layer_bounds)
+        reshaped_delta_p = delta_p.reshape(
+            (len(delta_p),) + (1,) * (cube_data.ndim - 1)
+        )
+        return reshaped_delta_p
+
+    @staticmethod
+    def calculate_precipitable_water(
+        sorted_data: ndarray, reshaped_delta_p: ndarray
+    ) -> ndarray:
+        """
+        Calculate total precipitable water using sorted humidity data and pressure thickness.
+
+        Args:
+            sorted_data:
+                Humidity mixing ratio data sorted by pressure levels.
+            reshaped_delta_p:
+                Pressure thickness values reshaped to match the data dimensions.
+
+        Returns:
+            Array of total precipitable water values in metres.
+        """
+        pw_data = (
+            sorted_data
+            * reshaped_delta_p
+            / (EARTH_SURFACE_GRAVITY_ACCELERATION * WATER_DENSITY)
+        )
+        pw_data = np.maximum(pw_data, 0)
+        total_pw_data = np.sum(pw_data, axis=0)
+        return total_pw_data
+
+    def sort_pressure_levels(self, cube: Cube) -> Tuple[ndarray, ndarray]:
+        """
+        Sort pressure levels in descending order and reorder the data accordingly.
+
+        Args:
+            cube:
+                Cube containing humidity mixing ratio data with a pressure coordinate.
+
+        Returns:
+            Tuple of sorted pressure levels and corresponding sorted data.
+
+        Raises:
+            ValueError: If the pressure coordinate is missing or not in Pascals.
+        """
+        try:
+            pressure_coord = cube.coord("pressure")
+        except iris.exceptions.CoordinateNotFoundError:
+            raise ValueError("Cube must have a 'pressure' coordinate.")
+
+        if pressure_coord.units != "Pa":
+            raise ValueError("Pressure units must be in Pascals (Pa).")
+
+        pressure_levels = pressure_coord.points
+        sorted_indices = np.argsort(pressure_levels)[::-1]
+        sorted_pressure = pressure_levels[sorted_indices]
+        sorted_data = cube.data[sorted_indices]
+
+        return sorted_pressure, sorted_data
+
+    def calculate_layer_bounds(self, sorted_pressure: ndarray) -> ndarray:
+        """
+        Calculate pressure layer boundaries from sorted pressure levels.
+
+        Args:
+            sorted_pressure:
+                1D array of pressure levels sorted in descending order.
+
+        Returns:
+            1D array of pressure layer boundaries.
+
+        Raises:
+            ValueError: If the resulting layer bounds are not strictly decreasing.
+        """
+        midpoints = 0.5 * (sorted_pressure[:-1] + sorted_pressure[1:])
+        layer_bounds = np.empty(len(sorted_pressure) + 1)
+        layer_bounds[1:-1] = midpoints
+        layer_bounds[0] = sorted_pressure[0] + (sorted_pressure[0] - midpoints[0])
+        layer_bounds[-1] = sorted_pressure[-1] - (midpoints[-1] - sorted_pressure[-1])
```

Comment:
This assumes that the lowest and highest pressure levels are only half as thick as other levels. I agree that this is the safe assumption, but I think you should describe this in one of the doc-strings.